src/lib/userStorage/FeedStorage.js

```diff
@@ -757,105 +749,27 @@ export class FeedStorage {
     }
   }
 
-  fromCeramicToStorage(ceramicPost) {
-    return {
-      id: ceramicPost.id,
-      createdDate: ceramicPost.published,
-      date: new Date(ceramicPost.published).getTime(),
-      displayType: 'news',
-      status: 'completed',
-      type: 'news',
-      data: {
-        message: ceramicPost.content,
-        subtitle: ceramicPost.title,
-        picture: ceramicPost.picture,
-      },
-    }
-  }
-
-  async getCeramicFeedPage(numResult: number) {
-    let ceramicPostsCached = await AsyncStorage.getItem('GD_CERAMIC_DATA')
-    log.debug('Ceramic cached posts', ceramicPostsCached)
-
-    if (!ceramicPostsCached) {
-      const ceramicPosts = await CeramicFeed.getPosts()
-      log.debug('Ceramic fetched posts', ceramicPosts)
-
-      const historyId = await CeramicFeed.getHistoryId()
-      await AsyncStorage.setItem('GD_CERAMIC_DATA', { posts: ceramicPosts, historyId })
-      ceramicPostsCached = { posts: ceramicPosts, historyId }
-    } else {
-      const { history, historyId } = await CeramicFeed.getHistory(ceramicPostsCached.historyId)
-      log.debug('Ceramic history', { history, historyId })
-
-      await Promise.all(
-        history.map(async ({ item: postId, action }) => {
-          log.debug('update feed item', { postId, action })
-
-          switch (action) {
-            case 'added': {
-              const post = await CeramicFeed.getPost(postId)
-              ceramicPostsCached.posts.push(post)
-              break
-            }
-            case 'updated': {
-              const post = await CeramicFeed.getPost(postId)
-              ceramicPostsCached.posts = ceramicPostsCached.posts.map(cachedItem =>
-                cachedItem.id === postId ? { ...cachedItem, post } : cachedItem,
-              )
-              break
-            }
-            default: {
-              ceramicPostsCached.posts = ceramicPostsCached.posts.filter(cachedItem => cachedItem.id !== postId)
-              break
-            }
-          }
-        }),
-      )
-
-      await AsyncStorage.setItem('GD_CERAMIC_DATA', {
-        ...ceramicPostsCached,
-        historyId,
-      })
-    }
-
-    const start = this.ceramicCursor
-    const end = this.ceramicCursor + numResult
-    return ceramicPostsCached.posts.slice(start, end).map(this.fromCeramicToStorage)
-  }
-
-  async getFeedPage(numResult: number, reset?: boolean) {
+  async getFeedPage(numResult: number, reset?: boolean, filterCallback?: (item: any) => boolean) {
     if (reset || isUndefined(this.cursor)) {
       this.cursor = 0
-      this.ceramicCursor = 0
     }
 
     await this.ready
 
-    const items = await this.storage.getFeedPage(numResult, this.cursor)
+    const items = await this.storage.getFeedPage(numResult, this.cursor, filterCallback)
```

Comment:
instead of filtering results by callback implement filters on the db query level please
(see `getFeedPage()` of RealmDB)